packages/miniflare/src/workers/kv/assets.worker.ts

```diff
@@ -13,21 +17,26 @@ export default <ExportedHandler<Env>>{
 			return new Response(message, { status: 405, statusText: message });
 		}
 
-		// Decode key
-		const url = new URL(request.url);
-		let key = url.pathname.substring(1); // Strip leading "/"
+		const decoder = new TextDecoder();
+		const reverseMap: AssetReverseMap = JSON.parse(
+			decoder.decode(env.__STATIC_ASSETS_REVERSE_MAP)
+		);
 
-		if (url.searchParams.get(KVParams.URL_ENCODED)?.toLowerCase() === "true") {
-			key = decodeURIComponent(key);
-		}
+		// don't uri decode pathname, because we encode the filepath before hashing
+		const key = new URL(request.url).pathname.substring(1);
```

Comment:
Let's be very explicit...

```suggestion
		const pathHash = new URL(request.url).pathname.substring(1);
```